packages/at_chops/lib/src/key/impl/at_chops_keys.dart

```diff
@@ -4,14 +4,29 @@ import 'package:at_chops/src/key/impl/at_pkam_key_pair.dart';
 import 'package:at_chops/src/key/impl/at_signing_key_pair.dart';
 
 class AtChopsKeys {
+  /// Default encryption key pair
   AtEncryptionKeyPair? atEncryptionKeyPair;
 
+  /// Key pair for pkam authentication. Can be legacy pkam keypair or apkam keypair for new enrollment
   AtPkamKeyPair? _atPkamKeyPair;
+
+  /// Key pair for data signing and verification
   AtSigningKeyPair? atSigningKeyPair;
+
+  @Deprecated('Use selfEncryptionKey')
   SymmetricKey? symmetricKey;
 
+  /// Default self encryption key
+  SymmetricKey? selfEncryptionKey;
```

Comment:
If the intent is that this replaces `symmetricKey` then shouldn't `symmetricKey` instance variable be replaced by a setter and getter for `symmetricKey` which use the `selfEncryptionKey` instance variable instead?

Reply:
my intent is to differentiate between self encryption key and apkam symmetric key.. If I keep a setter/getter symmetricKey, then the caller will be confused whether this is self encryption symmetric key or apkam symmetric key.

Reply:
Sorry my note wasn't clear.

You are adding `selfEncryptionKey`, which is "the same as" the current `symmetricKey`. You are marking `symmetricKey` as deprecated, but you are not removing it. As a result you have two instance variables which mean the same thing. But instead of keeping `symmetricKey` as is you could do this:
```
@Deprecated('Use selfEncryptionKey')
SymmetricKey? get symmetricKey => selfEncryptionKey;
@Deprecated('Use selfEncryptionKey')
void set symmetricKey(SymmetricKey? sk) => selfEncryptionKey = sk;
```